db/rdb.go

```diff
@@ -443,37 +436,22 @@ func (r *RDBDriver) CountNvd() (int, error) {
 
 // InsertNvd Cve information from DB.
 func (r *RDBDriver) InsertNvd(feedMetas []models.FeedMeta, cves map[string][]models.Nvd) (err error) {
-	logger.Infof("Inserting fetched CVEs...")
-	for _, meta := range feedMetas {
-		result := r.conn.Where(&models.FeedMeta{Source: meta.Source, Year: meta.Year}).Take(&meta)
-		if result.Error != nil && !errors.Is(result.Error, gorm.ErrRecordNotFound) {
-			return fmt.Errorf("Failed to select. err: %s", result.Error)
-		}
 
-		if result.RowsAffected == 0 {
-			if err := r.conn.Create(&meta).Error; err != nil {
-				return xerrors.Errorf("Failed to insert. err: %w", err)
-			}
-		}
-
-		if _, ok := cves[meta.Year]; !ok {
-			return xerrors.Errorf("Failed to get cves[%s].", meta.Year)
-		}
-
-		if err := r.deleteAndInsertNvd(r.conn, meta.ID, cves[meta.Year]); err != nil {
+	for y, cve := range cves {
+		logger.Infof("Inserting fetched CVEs(%s)...", y)
+		if err := r.deleteAndInsertNvd(r.conn, cve); err != nil {
 			return xerrors.Errorf("Failed to deleteAndInsertNvd. err: %w", err)
 		}
+	}
 
-		meta.Hash = meta.LatestHash
-		meta.LastModifiedDate = meta.LatestLastModifiedDate
-		if err := r.conn.Save(&meta).Error; err != nil {
-			return xerrors.Errorf("Failed to insert. err: %w", err)
-		}
+	logger.Infof("Updating FeedMetas...")
+	if err := r.upsertFeedMetas(feedMetas); err != nil {
+		return xerrors.Errorf("Failed to upsertFeedMetas. err: %w", err)
 	}
 	return nil
 }
 
-func (r *RDBDriver) deleteAndInsertNvd(conn *gorm.DB, feedMetaID uint, cves []models.Nvd) (err error) {
+func (r *RDBDriver) deleteAndInsertNvd(conn *gorm.DB, cves []models.Nvd) (err error) {
```

Comment:
Is there any reason not to use r.Conn?

Reply:
I don't know why, but I think they were wondering whether to implement it as an RDBDriver Method or a function.